packages/node/src/serverless-functions/serverless-handler.mts

```diff
@@ -21,40 +21,59 @@ type ServerlessFunctionSignature = (
   res: ServerResponse | VercelResponse
 ) => void;
 
-async function createServerlessServer(
-  userCode: ServerlessFunctionSignature,
-  options: ServerlessServerOptions
-) {
-  const server = createServer(async (req, res) => {
-    if (options.shouldAddHelpers) await addHelpers(req, res);
-    return userCode(req, res);
-  });
+const [NODE_MAJOR] = process.versions.node.split('.').map(v => Number(v));
+
+/* https://nextjs.org/docs/app/building-your-application/routing/router-handlers#supported-http-methods */
+const HTTP_METHODS = [
+  'GET',
+  'HEAD',
+  'OPTIONS',
+  'POST',
+  'PUT',
+  'DELETE',
+  'PATCH',
+];
+
+async function createServerlessServer(userCode: ServerlessFunctionSignature) {
+  const server = createServer(userCode);
   exitHook(() => server.close());
   return { url: await listen(server) };
 }
 
-async function compileUserCode(entrypointPath: string) {
+async function compileUserCode(
+  entrypointPath: string,
+  options: ServerlessServerOptions
+) {
   const id = isAbsolute(entrypointPath)
     ? pathToFileURL(entrypointPath).href
     : entrypointPath;
-  let fn = await import(id);
+  let listener = await import(id);
 
   /**
    * In some cases we might have nested default props due to TS => JS
    */
   for (let i = 0; i < 5; i++) {
-    if (fn.default) fn = fn.default;
+    if (listener.default) listener = listener.default;
+  }
+
+  if (HTTP_METHODS.some(method => typeof listener[method] === 'function')) {
+    if (NODE_MAJOR < 18) throw new Error('Node.js v18 or above is needed');
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** add detail to error

In this or a follow-up PR: Can we expand this error message to make it clearer what's going on?‏

```suggestion
    if (NODE_MAJOR < 18) {
      throw new Error('Node.js v18 or above is required to use HTTP method exports in your functions. Learn more: ${SOME_LINK}');
    }
```

Reply:
For now, we are going to just throw the error. When docs is prepared we can link it properly.

Reply:
This message is not rendered correctly as an error:

<img width="318" alt="image" src="https://github.com/vercel/vercel/assets/97262/e50f2f83-8ac7-49ab-abb6-3c71608c3cec">